File: src/main/java/fr/lewon/web/bot/methods/GetUserInfosMethod.java
package fr.lewon.web.bot.methods;

import java.util.ArrayList;
import java.util.List;

import fr.lewon.bot.errors.BotRunnerException;
import fr.lewon.bot.props.BotPropertyDescriptor;
import fr.lewon.bot.props.BotPropertyStore;
import fr.lewon.bot.runner.BotRunner;
import fr.lewon.web.bot.util.HHRequestProcessor;
import fr.lewon.web.bot.util.HHSessionManager;
import fr.lewon.web.bot.util.HtmlAnalyzer;

public class GetUserInfosMethod extends HHBotProcessor {

	public GetUserInfosMethod(String id, String label) {
		super(id, label, new ArrayList<>());
	}

	@Override
	protected List<BotPropertyDescriptor> getSpecificParamBuilders(BotRunner runner) {
		return new ArrayList<>();
	}

	@Override
	protected Object doProcess(BotRunner runner, HHSessionManager sessionManager, HHRequestProcessor requestProcessor,
			BotPropertyStore params) throws BotRunnerException {
		
		try {
			String homeContent = requestProcessor.getHomeContent(sessionManager.getSession(requestProcessor));
			return HtmlAnalyzer.INSTANCE.getPlayerInfos(homeContent);
		} catch (Exception e) {
			return e.getMessage();
		}
	}

}

File: src/main/kotlin/fr/lewon/bot/hh/operations/GetUserInfosMethod.kt
package fr.lewon.bot.hh.operations

import fr.lewon.bot.runner.Bot
import fr.lewon.bot.runner.bot.operation.BotOperation
import fr.lewon.bot.runner.bot.operation.OperationResult
import fr.lewon.bot.runner.bot.props.BotPropertyDescriptor
import fr.lewon.bot.runner.bot.props.BotPropertyStore


class GetUserInfosMethod : BotOperation("Get user infos") {

    override fun getNeededProperties(bot: Bot): List<BotPropertyDescriptor> {
        return ArrayList()
    }

    override fun run(bot: Bot, paramsPropertyStore: BotPropertyStore): OperationResult {
//        return try {
//            val homeContent = requestProcessor.getHomeContent(sessionManager.getSession(requestProcessor))
//            HtmlAnalyzer.INSTANCE.getPlayerInfos(homeContent)!!
//        } catch (e: Exception) {
//            e.message!!
//        }
        return OperationResult(false, null, null)
    }
}